src/ints/mouse.cpp

```diff
@@ -509,76 +559,22 @@ void Mouse_CursorMoved(float xrel,float yrel,float x,float y,bool emulate) {
 		if (mouse.y >= 32768.0) mouse.y -= 65536.0;
 		else if (mouse.y <= -32769.0) mouse.y += 65536.0;
 	}
-	Mouse_AddEvent(MOUSE_HAS_MOVED);
+	AddEvent(DOS_EV::MOUSE_HAS_MOVED);
 	DrawCursor();
 }
 
-void Mouse_CursorSet(float x,float y) {
-	mouse.x=x;
-	mouse.y=y;
-	DrawCursor();
-}
-
-void Mouse_ButtonPressed(uint8_t button) {
-	switch (button) {
-#if (MOUSE_BUTTONS >= 1)
-	case 0:
-		if (mouse.buttons&1) return;
-		mouse.buttons|=1;
-		Mouse_AddEvent(MOUSE_LEFT_PRESSED);
-		break;
-#endif
-#if (MOUSE_BUTTONS >= 2)
-	case 1:
-		if (mouse.buttons&2) return;
-		mouse.buttons|=2;
-		Mouse_AddEvent(MOUSE_RIGHT_PRESSED);
-		break;
-#endif
-#if (MOUSE_BUTTONS >= 3)
-	case 2:
-		if (mouse.buttons&4) return;
-		mouse.buttons|=4;
-		Mouse_AddEvent(MOUSE_MIDDLE_PRESSED);
-		break;
-#endif
-	default:
-		return;
-	}
-	mouse.times_pressed[button]++;
-	mouse.last_pressed_x[button]=POS_X;
-	mouse.last_pressed_y[button]=POS_Y;
+static inline uint8_t GetResetWheel8bit() {
+    if (!mouse.cute_mouse) return 0;
+    int8_t tmp = static_cast<int8_t>(std::clamp(mouse.wheel, static_cast<int16_t>(-0x80), static_cast<int16_t>(0x7f)));
+    mouse.wheel = 0;
+    return (tmp >= 0) ? tmp : 0x100 + tmp;
```

Comment:
Maybe a short comment would be helpful to explain what you're doing here with adding the offset.